examples/v1beta1/taskruns/alpha/emit-array-results.yaml

```diff
@@ -0,0 +1,30 @@
+kind: Task
```

Comment:
In addition to the `Task`, please add an example `Pipeline`. The results are more of a communication means between two tasks (or pipelineTasks). The example are great way to share the usage of the new features.

Reply:
Hi @pritidesai, thanks for your comment! 
In this PR array results passed between tasks are not implemented. This commit focuses on enabling emitting array results from entrypoint->controller. 
I'm planning to add it (also array indexing) as well as examples in the followup PR.  
Or should I add it in this PR?  What do you think?

Reply:
Or maybe I should remove this example and add both examples for next pr?

Reply:
Thank you @Yongxuanzhang for the explanation. 

> In this PR array results passed between tasks are not implemented. 

You mean just the indexing is not supported in this PR or in general, the consumer will get an array result as a string value? I am trying to understand what happens when a consumer refers to such array result in a traditional way i.e. `$tasks.taskA.results.arrayResult`.

Reply:
So with this PR, if referred `$tasks.taskA.results.arrayResult` in params and it is an array, the users will get a string of `$tasks.taskA.results.arrayResult` not the array or the string format of the array. 

It is also something related to this [issue](https://github.com/tektoncd/pipeline/issues/4879). 
This is a bit tricky part and it would be better to be addressed in a separate PR. Since `$tasks.taskA.results.arrayResult` or `$tasks.taskA.results.arrayResult[*]` will be treated as `string` type in `ArrayOrString`.

Reply:
Oh I realized that the doc I added may be confusing as well, support emitting array results may let users think that they can pass array results between tasks within a pipeline, but it is not implemented in this PR.

Reply:
> So this PR only enables the Task level results array, not sure if this is a real case.

It might be or might not be, we do not advice such usage but its not restricted either.

Reply:
Should I fix the readme to clarify this? @pritidesai

Reply:
thank you @Yongxuanzhang I added comment ⬆️

Reply:
I have added more clarification in the doc